server.py

```
@@ -304,6 +304,36 @@ def _analyze_comment_impact(score: int, is_edited: bool, is_op: bool) -> str:
 
+def _serialize_comment_tree(comment: praw.models.Comment) -> Dict[str, Any]:
+    """Serialize a PRAW comment into a JSON-serializable tree structure."""
+    try:
+        replies = []
+        if getattr(comment, "replies", None):
+            replies = [
+                _serialize_comment_tree(reply)
+                for reply in comment.replies
+                if isinstance(reply, praw.models.Comment)
+            ]
+    except Exception as e:
```

Ruff (0.14.8)

Do not catch blind exception: `Exception`

(BLE001)